setup.py

```diff
@@ -63,14 +63,14 @@ def build_expected_version_hex(matches):
 )
 
 # Read the listed version
-with open("pybind11/_version.py") as f:
+with open("pybind11/_version.py", encoding="utf8") as f:
```

Comment:
This needs to be io.open for Python2

Reply:
Unfortunately, that gives `SyntaxError: encoding declaration in Unicode string` in python2 package building.

Reply:
Something like this seems to work, but is a somewhat ugly solution, i think:

```python
encoding_for_py3 = {"encoding" : "utf8"} if sys.version_info.major >= 3 else {}
with open("pybind11/_version.py", **encoding_for_py3) as f:
```

Reply:
Are you sure you need encoding here? I think it's correct below, but this one is being exec'd, so it can pick up the encoding from the file. Maybe it could even be opened explicitly in binary?